src/rgw/driver/rados/rgw_data_sync.cc

```diff
@@ -1852,12 +1852,11 @@ class RGWDataFullSyncShardCR : public RGWDataBaseSyncShardCR {
       omapvals.reset();
 
       drain_all();
+      yield call(marker_tracker->flush());
```

Comment:
this adds a `flush()` in 2 cases where it wasn't before:

when `lost_lock` is true, that means some other radosgw has acquired it, read the sync status, and started sync. if we issue a racing write to the sync status here, that'll cause the other radosgw to fail with `ECANCELED` when it wants to record its own progress

when `lost_lock` is false, we're overwriting the sync status anyway to switch to `IncrementalSync`, so the `flush()` is wasted there

Reply:
hmm but then how do we handle the case of `lost_lock`. don't flush at all?
in incremental sync, we do seem to call flush() in all cases.

Reply:
> how do we handle the case of lost_lock. don't flush at all?

right. the `drain_all()` means that all the 'sync single entry' coroutines have completed, so nothing else would try to flush. if another radosgw has the lock, it should be responsible for managing the sync status

> in incremental sync, we do seem to call flush() in all cases

i feel like incremental sync should avoid the `flush()` on `lost_lock` for the same reason

Reply:
great, thanks for elaborating!